dspace-api/src/main/java/org/dspace/app/mediafilter/PDFBoxThumbnail.java

```diff
@@ -70,10 +70,26 @@ public InputStream getDestinationStream(Item currentItem, InputStream source, bo
         throws Exception {
         BufferedImage buf;
 
+        String pageNumberStr =  currentItem.getItemService()
+            .getMetadataFirstValue(currentItem, "dspace", "filter-media", "page", Item.ANY);
```

Comment:
This new field needs to be added to the `dspace` internal schema, along with a note that it is a zero-based index.

I'd also like to add that we may wish to consider renaming this to be `dspace.thumbnail.page` because it's *only used for thumbnails* and not all other filter-media scripts (like full text indexing, etc).

Finally, it seems like it may be a very small change to update `ImageMagickThumbnailFilter` to **also use** this setting.  It already has a specified page number which is hardcoded to "0" here: https://github.com/DSpace/DSpace/blob/main/dspace-api/src/main/java/org/dspace/app/mediafilter/ImageMagickThumbnailFilter.java#L157   Should we consider updating this PR to also change `ImageMagickThumbnailFilter`?

Reply:
You are right, I have changed the metadata name to `dspace.thumbnail.page`.
Also, I have added code to get the value of the same metadata and take care of it when using the IM thumbnail filter.